src/character/player.rs

```diff
@@ -1,56 +1,252 @@
+    replication::transform::SyncedTransform,
+};
 
 use super::{CharacterPhysicsBundle, CharacterVectors};
 
 pub const RADIUS: f32 = 0.4;
 pub const HALF_HEIGHT: f32 = 0.4;
 
-#[derive(Component)]
-pub struct PlayerControls;
+pub struct PlayerPlugin;
+
+impl Plugin for PlayerPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_server_event::<TransformEvent>(EventType::Ordered)
+            .add_client_event::<TransformEvent>(EventType::Ordered)
+            .replicate::<Player>()
+            .add_systems(
+                PreUpdate,
+                (
+                    init_players.after(ClientSet::Receive).run_if(has_client()),
+                    transform_server_handler
+                        .after(ServerSet::Receive)
+                        .run_if(has_server()),
+                    transform_client_handler
+                        .after(ClientSet::Receive)
+                        .run_if(has_client()),
+                ),
+            )
+            .add_systems(
+                FixedUpdate,
+                (
+                    control.run_if(has_local_player()),
+                    transform_server_sender.run_if(has_server()),
+                    transform_client_sender.run_if(has_client_and_local()),
+                ),
+            );
+    }
+}
+
+pub fn spawn(
+    commands: &mut Commands,
+    meshes: &mut ResMut<Assets<Mesh>>,
+    materials: &mut ResMut<Assets<StandardMaterial>>,
+    player: Player,
+    is_local: bool,
+) {
+    let transform = Transform::from_xyz(0.0, 3.0, 0.0);
+    let mut c = commands.spawn((
+        player,
+        SharedPlayerBundle::new(meshes, materials, transform, is_local),
+        Ignored::<Transform>::default(),
+        Ignored::<CharacterVectors>::default(),
+    ));
+
+    if is_local {
+        c.insert(LocalPlayerBundle::default());
+        commands.insert_resource(LocalPlayerResource);
+    } else {
+        c.insert(RemotePlayerBundle {
+            synced_transform: transform.into(),
+        });
+    }
+}
+
+fn init_players(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    spawned_players: Query<(Entity, &Player), Added<Player>>,
+    client: Res<Client>,
+) {
+    for (entity, player) in &spawned_players {
+        let transform = Transform::from_xyz(0.0, 3.0, 0.0);
+        let mut c = commands.entity(entity);
+        c.insert(SharedPlayerBundle::new(
+            &mut meshes,
+            &mut materials,
+            transform,
+            player.client_id == client.id,
+        ));
+
+        if player.client_id == client.id {
+            c.insert(LocalPlayerBundle::default());
+            commands.insert_resource(LocalPlayerResource);
+        } else {
+            c.insert(RemotePlayerBundle {
+                synced_transform: transform.into(),
+            });
+        }
+    }
+}
+
+#[derive(Component, Serialize, Deserialize)]
+pub struct Player {
+    pub client_id: u64,
```

Comment:
I would make this `client_id` into a newtype, since a plain `u64` is easy to confuse with other `u64`s.

```rs
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub struct ClientId(u64);
```

That way it is strongly typed and cannot be passed around where a plain `u64` is expected, as well as denying any math operations that wouldn't make sense on unique identifiers (such as addition, multiplication, etc.)